xrchat/xrchat-client
# Upgraded user profile tab.

**What is the purpose of this PR?**
Updated the user profile tab so that it actually uploads and displays the user's avatar.
Added a field to display and update the user's name.

**What does issue #s are solved by this PR?**
No specific issues

**What unit tests have been put in place?**
None

**What files or features are including in this PR that are not related to the main feature?**
None

**How does our QA team test this PR?**
This must be tested with the server branch 'user-profile-fixes'.

Login with a user. You'll see that the placeholder letter in the upper right is actually the first letter of their name, not always X.

Click on that to open the menu, then click Profile. The leftmost tab is now called 'Profile', and you'll see it now has a text field with their username and a button to update their username.

If you change the username and press the button, there should be a notification that their username was updated. If you reload the page and go back to that tab, the name should be what you updated it to.

The avatar uploading is improved and functional. If you hover over the placeholder avatar image, the cursor will be a pointer and the background will highlight to indicate that that's a clickable area. The Upload Avatar button should now be disabled if no new image is selected.

If you click on the dummy avatar, select an image, then click Upload Avatar, after a few seconds there should be a notification that the avatar was updated, and the avatar in the upper right corner of the screen (where you click to open the dropdown) should now display the new avatar.
